k9mail/src/main/java/com/fsck/k9/provider/MailListViewWidgetProvider.java

```diff
@@ -0,0 +1,124 @@
+package com.fsck.k9.provider;
+
+import android.app.PendingIntent;
+import android.appwidget.AppWidgetManager;
+import android.appwidget.AppWidgetProvider;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.widget.RemoteViews;
+import com.fsck.k9.Preferences;
+import com.fsck.k9.R;
+import com.fsck.k9.activity.MailListViewWidgetConfiguration;
+import com.fsck.k9.activity.MessageCompose;
+import com.fsck.k9.activity.MessageList;
+import com.fsck.k9.service.MailListViewWidgetService;
+
+/**
+ * Implementation of App Widget functionality.
+ */
+public class MailListViewWidgetProvider extends AppWidgetProvider {
+    public static final String PACKAGE_NAME = "com.fsck.k9";
+
+    public static String ACTION_VIEW_MAIL_ITEM = PACKAGE_NAME + ".provider.ACTION_VIEW_MAIL_ITEM";
+    public static String ACTION_COMPOSE_EMAIL = PACKAGE_NAME + ".provider.ACTION_COMPOSE_EMAIL";
+
+
+    private static String account;
+
+    /**
+     * When a new email is received or deleted call this method to broadcast an intent
+     * in order to update the listview of the widget
+     * @param context
+     */
+    public static void updateMailViewList(Context context) {
+        Context appContext = context.getApplicationContext();
+        AppWidgetManager widgetManager = AppWidgetManager.getInstance(appContext);
+        ComponentName widget = new ComponentName(appContext, MailListViewWidgetProvider.class);
+        int [] widgetIds = widgetManager.getAppWidgetIds(widget);
+
+        Intent intent = new Intent(context, MailListViewWidgetProvider.class);
+        intent.setAction(AppWidgetManager.ACTION_APPWIDGET_UPDATE);
+        intent.putExtra(AppWidgetManager.EXTRA_APPWIDGET_IDS, widgetIds);
+        context.sendBroadcast(intent);
+    }
+
+    public static void updateAppWidget(Context context, AppWidgetManager appWidgetManager, int appWidgetId, String acc) {
+
+        CharSequence widgetText = context.getString(R.string.appwidget_text);
+        // Construct the RemoteViews object
+        RemoteViews views = new RemoteViews(context.getPackageName(), R.layout.mail_list_view_widget_layout);
+        views.setRemoteAdapter(R.id.listView, new Intent(context, MailListViewWidgetService.class));
+        views.setTextViewText(R.id.folder, "INBOX");
+        Intent intent = new Intent(context, MailListViewWidgetService.class)
+                .putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, appWidgetId);
+        intent.setData(Uri.parse(intent.toUri(Intent.URI_INTENT_SCHEME)));
+        views.setRemoteAdapter(R.id.listView, intent);
+
+        Intent clickIntent = new Intent(context, MailListViewWidgetProvider.class);
+        clickIntent.setPackage(PACKAGE_NAME);
+        clickIntent.setAction(ACTION_VIEW_MAIL_ITEM);
+        PendingIntent pendingIntent = PendingIntent.getBroadcast(context, MessageList.REQUEST_MASK_PENDING_INTENT,
```

Comment:
I've tried using PendingIntent#getActivity() method as well but nothing happened no action what so ever when i clicked on the widget.
This is due to the way you set up behaviour for individual items in a widget's collection.[more here](https://developer.android.com/guide/topics/appwidgets/index.html#implementing_collections)
The main points are that
* You need to set a pending intent template for the entire collection
* add fill intent for each item (in getViewAt method), the fill in intent is combined with the pendingIntent 
* process the event in the onReceive method of AppWidgetProvider class.